python/ray/serve/_private/deployment_state.py

```diff
@@ -250,6 +250,8 @@ def __init__(
         self._initialization_latency_s: Optional[float] = None
         self._port: Optional[int] = None
         self._docs_path: Optional[str] = None
+        # Rank assigned to the replica.
+        self._rank: Optional[int] = None
```

Comment:
why is rank optional and initially set to None? if it's something assigned by the controller we should be able to know the rank on init of this object right

Reply:
but rank is assigned when .start is invoked not when this object is created.

Reply:
right, I'm asking if there is a reason we populate it in `.start` instead of when the object is created

Reply:
ah, good point. It doesn't need to be on start. Will change the implementation